lib/services/aiSynthesis/index.ts

```diff
@@ -220,22 +259,90 @@ class AISynthesis extends AbstractManager {
       },
     }));
 
-    if (!synthesis) return { ...EMPTY_AI_RESPONSE, chunks };
+    if (!synthesis) return of({ ...EMPTY_AI_RESPONSE, chunks });
 
-    const answer = await this.services.aiSynthesis.answerSynthesis({
-      question,
-      instruction,
-      data,
-      options: settings?.summarization,
-      context: { projectID: project._id, workspaceID: project.teamID },
-    });
+    const stream$ = new BufferedReducerSubject<KBResponse>(
+      (resp) => {
+        const output = (resp.output ?? '').toLocaleUpperCase();
+        // Stop stream all together if output is a bad phrase
+        if (this.filterNotFound(resp.output ?? '') === null) return 'stop';
+
+        // Keep buffering if the output appears to match a bad phrase
+        return NOT_FOUND_RESPONSES.some((phrase) => phrase.startsWith(output));
+      },
+      (buffer, value) => {
+        if (!buffer.output) buffer.output = '';
+        if (!buffer.chunks) buffer.chunks = [];
+
+        buffer.chunks.push(...(value.chunks ?? []));
+        buffer.output += value.output ?? '';
+        buffer.answerTokens += value.answerTokens;
+        buffer.queryTokens += value.queryTokens;
+        buffer.tokens += value.tokens;
+        buffer.model = value.model;
+        buffer.multiplier = value.multiplier;
+        buffer.faqSet = value.faqSet;
+        return buffer;
+      }
```

Comment:
lets keep this custom Subject usage as immutable as possible, avoid mutate methods like `.push` and opt for spread, `concat`, `reduce`, etc
```suggestion
      (buffer, value) => ({
        chunks: (buffer.chunks || []).concat(value.chunks ?? []),
        output: (buffer.output ?? "") + (value.output ?? ""),
        answerTokens: buffer.answerTokens + value.answerTokens,
        queryTokens: buffer.queryTokens + value.queryTokens,
        tokens: buffer.tokens + value.tokens,
        model: value.model,
        multiplier: value.multiplier,
        faqSet: value.faqSet,
      })
```